tests/mock_crust/churn.rs

```diff
@@ -201,64 +317,127 @@ fn verify_section_list_signatures(nodes: &[TestNode]) {
 
 #[test]
 fn churn() {
+    let min_section_size = 5;
+    let mut network = Network::new(min_section_size, None);
+    let mut rng = network.new_rng();
+
+    // Create an initial network, increase until we have several sections, then
+    // decrease back to min_section_size, then increase to again.
+    let mut nodes = create_connected_nodes(&network, min_section_size);
+
+    info!("Churn [{} nodes, {} sections]: adding nodes",
+          nodes.len(),
+          count_sections(&nodes));
+    loop {
+        let (added_index, _) = add_random_node(&mut rng, &network, &mut nodes, min_section_size);
+        poll_and_resend(&mut nodes, &mut []);
+        send_and_receive(&mut rng, &mut nodes, min_section_size, Some(added_index));
+        if count_sections(&nodes) > 5 {
+            break;
+        }
+    }
+
+    info!("Churn [{} nodes, {} sections]: dropping nodes",
+          nodes.len(),
+          count_sections(&nodes));
+    while nodes.len() > min_section_size {
+        drop_random_nodes(&mut rng, &mut nodes, min_section_size);
+        poll_and_resend(&mut nodes, &mut []);
+        send_and_receive(&mut rng, &mut nodes, min_section_size, None);
+        client_gets(&mut network, &mut nodes, min_section_size);
+    }
+
+    info!("Churn [{} nodes, {} sections]: adding nodes",
+          nodes.len(),
+          count_sections(&nodes));
+    while nodes.len() < 50 {
+        let (added_index, _) = add_random_node(&mut rng, &network, &mut nodes, min_section_size);
+        poll_and_resend(&mut nodes, &mut []);
+        send_and_receive(&mut rng, &mut nodes, min_section_size, Some(added_index));
+        client_gets(&mut network, &mut nodes, min_section_size);
+    }
+
+    info!("Churn [{} nodes, {} sections]: simultaneous adding and dropping nodes",
+          nodes.len(),
+          count_sections(&nodes));
+    for _ in 0..10 {
```

Comment:
I think this can be changed for `while nodes.len() > min_section_size + 1` since we always remove more nodes than we add.  We should eventually remove so many that `nodes` goes down to `min_section_size` then we add another one.  We can also then remove the 3 following lines.